main.go

```diff
@@ -1,506 +1,187 @@
+// the uploaded apps.
+func uploadApplications(configs Configs, service *androidpublisher.Service, appEdit *androidpublisher.AppEdit) ([]int64, error) {
+	var versionCodes []int64
+	appPaths, _ := configs.AppPaths()
 
-	config, err := google.JWTConfigFromJSON(jsonKeyBytes, androidpublisher.AndroidpublisherScope)
+	expansionFileUpload, expansionFilePaths, err := GetExpansionFiles(appPaths, configs.ExpansionfilePath)
 	if err != nil {
-		return nil, err
+		return []int64{}, err
 	}
 
-	return config, nil
-}
-
-func jwtConfigFromP12KeyFile(pth, email string) (*jwt.Config, error) {
-	cmd := command.New("openssl", "pkcs12", "-in", pth, "-passin", "pass:notasecret", "-nodes")
+	for i, appPath := range appPaths {
+		log.Printf("Uploading %v %d/%d", appPath, i+1, len(appPaths))
+		versionCode := int64(0)
+		appFile, err := os.Open(appPath)
+		if err != nil {
+			return []int64{}, fmt.Errorf("failed to open app (%s), error: %s", appPath, err)
+		}
 
-	var outBuffer bytes.Buffer
-	outWriter := bufio.NewWriter(&outBuffer)
-	cmd.SetStdout(outWriter)
+		if strings.ToLower(filepath.Ext(appPath)) == ".aab" {
+			bundle, err := UploadAppBundle(service, configs.PackageName, appEdit.Id, appFile)
+			if err != nil {
+				return []int64{}, err
+			}
+			versionCodes = append(versionCodes, bundle.VersionCode)
+			versionCode = bundle.VersionCode
+		} else {
+			apk, err := UploadAppApk(service, configs.PackageName, appEdit.Id, appFile)
+			if err != nil {
+				return []int64{}, err
+			}
+			versionCodes = append(versionCodes, apk.VersionCode)
+			versionCode = apk.VersionCode
 
-	var errBuffer bytes.Buffer
-	errWriter := bufio.NewWriter(&errBuffer)
-	cmd.SetStderr(errWriter)
+			if expansionFileUpload {
+				if err := UploadExpansionFiles(service, expansionFilePaths[i], configs.PackageName, appEdit.Id, versionCode); err != nil {
+					return []int64{}, err
+				}
+			}
+		}
 
-	if err := cmd.Run(); err != nil {
-		if !errorutil.IsExitStatusError(err) {
-			return nil, err
+		// Upload mapping.txt
+		if configs.MappingFile != "" && versionCode != 0 {
+			if err := UploadMappingFile(service, configs, appEdit.Id, versionCode); err != nil {
+				return []int64{}, err
+			}
+			if i < len(appPaths)-1 {
+				fmt.Println()
+			}
 		}
-		return nil, errors.New(string(errBuffer.Bytes()))
 	}
-
-	return &jwt.Config{
-		Email:      email,
-		PrivateKey: outBuffer.Bytes(),
-		TokenURL:   google.JWTTokenURL,
-		Scopes:     []string{androidpublisher.AndroidpublisherScope},
-	}, nil
+	log.Printf("Done uploading of %v apps", len(appPaths))
+	log.Printf("New version codes to upload: %v", versionCodes)
+	return versionCodes, nil
 }
 
-func readLocalisedRecentChanges(recentChangesDir string) (map[string]string, error) {
-	recentChangesMap := map[string]string{}
+// updateTracks updates the given track with a new release with the given version codes.
+func updateTracks(configs Configs, service *androidpublisher.Service, appEdit *androidpublisher.AppEdit, versionCodes []int64) error {
```

Comment:
Could we order versionCodes by a specific order before passing it to this function? (By ordering the list of apks alphabetivally for example.) The order is dependent on the order of parsed Apks, but that could change any time. The order seems that is used later to get the latest versions of the specific apks (the unblock versions logic).